File: src/main/java/com/kingmang/lazurite/parser/AST/Statements/ContinueStatement.java
package com.kingmang.lazurite.parser.AST.Statements;


import com.kingmang.lazurite.patterns.visitor.ResultVisitor;
import com.kingmang.lazurite.patterns.visitor.Visitor;

public final class ContinueStatement extends RuntimeException implements Statement {

    @Override
    public void execute() {
        throw this;
    }
    
    @Override
    public void accept(Visitor visitor) {
        visitor.visit(this);
    }

    @Override
    public <R, T> R accept(ResultVisitor<R, T> visitor, T t) {
        return visitor.visit(this, t);
    }

    @Override
    public String toString() {
        return "continue";
    }
}

File: src/main/java/com/kingmang/lazurite/parser/ast/statements/ContinueStatement.kt
package com.kingmang.lazurite.parser.ast.statements

import com.kingmang.lazurite.patterns.visitor.ResultVisitor
import com.kingmang.lazurite.patterns.visitor.Visitor

class ContinueStatement : RuntimeException(), Statement {
    override fun execute() =
        throw this

    override fun accept(visitor: Visitor) =
        visitor.visit(this)

    override fun <R, T> accept(visitor: ResultVisitor<R, T>, input: T): R? =
        visitor.visit(this, input)

    override fun toString(): String =
        "continue"
}
